tests/compiler/optimize-instruction-regressions.optimized.wat

```diff
@@ -0,0 +1,48 @@
+(module
+ (type $none_=>_none (func))
+ (type $i32_i32_i32_i32_=>_none (func (param i32 i32 i32 i32)))
+ (import "env" "abort" (func $~lib/builtins/abort (param i32 i32 i32 i32)))
+ (memory $0 1)
+ (data (i32.const 16) "F\00\00\00\01\00\00\00\01\00\00\00F\00\00\00o\00p\00t\00i\00m\00i\00z\00e\00-\00i\00n\00s\00t\00r\00u\00c\00t\00i\00o\00n\00-\00r\00e\00g\00r\00e\00s\00s\00i\00o\00n\00s\00.\00t\00s\00")
+ (table $0 1 funcref)
+ (global $optimize-instruction-regressions/x (mut i32) (i32.const 128))
+ (export "memory" (memory $0))
+ (start $~start)
+ (func $start:optimize-instruction-regressions
+  global.get $optimize-instruction-regressions/x
+  i32.const 24
+  i32.shl
+  i32.const 24
+  i32.shr_u
+  i32.const -128
+  i32.ne
```

Comment:
Original binaryen don't change this from `shr_s` to `shr_u`. binaryen.js behave differently. Need farther inverstigation